src/js/view/floating/floating-controller.ts

```diff
@@ -151,6 +162,7 @@ export default class FloatingController {
         if (this.getFloatingPlayer() === null) {
             this.setFloatingPlayer(this._playerEl);
 
+            preTransition(this);
             this._model.set('isFloating', true);
 
             addClass(this._playerEl, 'jw-flag-floating');
```

Comment:
`preTransition` should be placed next to where `'jw-flag-floating'` is added and removed since that is what triggers the transition.